src/main/java/RecommendationsServlet.java

```diff
@@ -0,0 +1,184 @@
+  // US state in 2018. 
+  private ArrayList<VisualizationData> getDefaultRecommendations(String userId) {
+    ArrayList<VisualizationData> defaultRecommendations = new ArrayList<VisualizationData>();
+    for (String person: personType) {
+      defaultRecommendations.add(new VisualizationData(userId, person, "live", "state", "2018"));
+    }
+    return defaultRecommendations;
+  }
+
+  // uses java.util.random to get a random integer between 
+  // 0 and upperBound-1 in order to access a random element 
+  // for each of the options
+  private int getRandomInt(int upperBound) {
+    Random randomObject = new Random();
+    int randomInt = randomObject.nextInt(upperBound);
+    return randomInt;
+  }
+
+  // CHeck to see if the current recommendation is valid by making sure that it isn't an impossible combination 
+  // and isn't already one that the user has viewed.
+  private boolean isRecommendationValid(VisualizationData recommendation, ArrayList<VisualizationData> userHistory) {
+      return (!(userHistory.contains(recommendation) || 
+        (recommendation.getPersonType() == "under-18" && recommendation.getAction() == "work") ||
+        (recommendation.getPersonType() == "under-18" && recommendation.getAction() == "moved"))); 
+  }
+
+  // Returns a VisualizationData whose contents are completely random
+  // in order to alleviate recommendation fatigue. 
+  private VisualizationData getRandomRecommendation(String userId, ArrayList<VisualizationData> userHistory) {
+    ArrayList<VisualizationData> invalidRecommendations = new ArrayList<VisualizationData>();
+    boolean isValid = false;
+    while (!isValid) {
+      String randomPersonType = personType.get(getRandomInt(personType.size()));
+      String randomAction = action.get(getRandomInt(action.size()));
+      String randomLocation = location.get(getRandomInt(location.size()));
+      String randomYear = years.get(getRandomInt(years.size()));
+      VisualizationData randomRecommendation = new VisualizationData(
+          userId, randomPersonType, randomAction, randomLocation, randomYear);
+      if (isRecommendationValid(randomRecommendation, userHistory) && !invalidRecommendations.contains(randomRecommendation)) {
+        return randomRecommendation;
+      } else {
+        invalidRecommendations.add(randomRecommendation);
+      }
+      // If the user has looked at every visualization, return the first in the history.
+      if (invalidRecommendations.size() >= 6885) {
+        isValid = true;
+      }
+    }
+    return userHistory.get(0);
+  }
+
+  // Takes in a map and updates the key/value pairs
+  private void updateMap(HashMap<String, Integer> frequencyMap, String item) {
+    if (frequencyMap.containsKey(item)) {
+      frequencyMap.put(item, frequencyMap.get(item) + 1);
+    } else {
+      frequencyMap.put(item, 1);
+    }
+  }
+
+  private void getFrequencyMaps(ArrayList<VisualizationData> userHistory) {
```

Comment:
Maybe this should be called updateFrequencyMaps as I don't see any gets being done here.

Reply:
Changed to updateFrequencyMaps